packages/react/src/use-feature-flag.ts

```diff
@@ -0,0 +1,47 @@
+import { Client, EvaluationDetails, FlagValue, ProviderEvents, ResolutionDetails } from '@openfeature/web-sdk';
+import { useEffect, useState } from 'react';
+import { useOpenFeatureClient } from './provider';
+
+export function useFeatureFlag<T extends FlagValue>(flagKey: string, defaultValue: T): T {
+  const [flagEvaluationDetails, setFlagEvaluationDetails] = useState<ResolutionDetails<T> | null>(null);
+
+  const client = useOpenFeatureClient();
+
+  useEffect(() => {
+    getFlag(client, flagKey, defaultValue, setFlagEvaluationDetails);
+    // adding handlers here mean that changes are immediately reflected in the UI when flags change
+    client.addHandler(ProviderEvents.Ready, () => {
```

Comment:
I believe we also want to add `client.removeHandler` calls in the [cleanup method](https://blog.logrocket.com/understanding-react-useeffect-cleanup-function/) that we return to `useEffect`

Reply:
[Great minds.](https://github.com/open-feature/js-sdk/pull/546#discussion_r1301935549)